src/kbucket.rs

```diff
@@ -51,61 +54,61 @@ impl KbucketTable {
         let result = self.search_table(id);
-        let mut bucket = self.buckets[result.bucket_index];
-
-        if result.found {
-            // Returns Node
-            FindNodeResult::Found(bucket[result.column_index])
-        } else {
-            let mut known_nodes = Vec::new();
 
-            for node in bucket.iter() {
-                if node.is_some() {
-                    // Should I be dereferencing the node to send to others?  Or copy the node to share?
-                    known_nodes.push(*node)
+        match result {
+            Search::Success(bucket_index, column_index) => {
+                let bucket = self.buckets[bucket_index];
+                FindNodeResult::Found(bucket[column_index])
+            }
+            Search::Failure(bucket_index, column_index) => {
+                let bucket = self.buckets[bucket_index];
+                let mut known_nodes = Vec::new();
+
+                for node in bucket.iter() {
+                    if node.is_some() {
+                        // Should I be dereferencing the node to send to others?  Or copy the node to share?
+                        known_nodes.push(*node)
+                    }
                 }
+                FindNodeResult::NotFound(known_nodes)
             }
-            // Returns nodes within local node's closest bucket to queried node
-            FindNodeResult::NotFound(known_nodes)
         }
     }
     // TODO:
     pub fn find_value() {}
 
     // TODO:
-    /// Instructs a node to store a key, value pair for later retrieval. "Most operations are implemented
-    /// in terms of the lookup proceedure. To store a <key,value> pair, a participant locates the k closes
-    /// nodes to the key and sends them store RPCs".
+    /// Instructs a node to store a key, value pair for later retrieval.
+    ///
+    /// "Most operations are implemented in terms of the lookup proceedure. To store a
+    /// <key,value> pair, a participant locates the k closes nodes to the key and sends them store RPCs".
     pub fn store(&mut self, key: Identifier, value: Vec<u8>) {}
 
     // Non-RPCs:
     // ---------------------------------------------------------------------------------------------------
-    pub fn add_node(&mut self, node: Node) {
+    // TODO:  Make add_node() take a reference to a node, not a node directly
+    fn add_node(&mut self, node: &Node) -> bool {
         let result = self.search_table(node.node_id);
-        let mut bucket = self.buckets[result.bucket_index];
-
-        if !result.found {
-            bucket[result.column_index] = Some(node)
-        } else {
-            println!("Node is already in our table")
+        match result {
+            Search::Success(bucket_index, column_index) => false,
+            Search::Failure(bucket_index, column_index) => {
+                // Should I be dereferencing the node?  Or change buckets to take references to nodes?
+                self.buckets[bucket_index][column_index] = Some(*node);
```

Comment:
yes just deref ("copy") or even clone the node here

there are more sophisticated things you could end up doing but this is out of scope for now